hudi-common/src/main/java/org/apache/hudi/common/config/SerializableSchema.java

```diff
@@ -62,12 +62,17 @@ private void readObject(ObjectInputStream in) throws IOException {
   
   // create a public write method for unit test
   public void writeObjectTo(ObjectOutputStream out) throws IOException {
-    out.writeUTF(schema.toString());
+    // Note: writeUTF cannot support string length > 64K. So use writeObject which has small overhead (relatively).
+    out.writeObject(schema.toString());
   }
 
   // create a public read method for unit test
   public void readObjectFrom(ObjectInputStream in) throws IOException {
-    schema = new Schema.Parser().parse(in.readUTF());
+    try {
+      schema = new Schema.Parser().parse(in.readObject().toString());
```

Comment:
Can we do the following : 

int length = in.readInt();
byte[] value = new byte[length];
in.readFully(data);
String schemaStr = new String(value, "UTF-8");

This will ensure UTF encoding/decoding

Reply:
This was initial implementation. Unfortunately, it doesnt well with java serialization because we have 'out.defaultWriteObject();' in front.  Per Vinoth's suggestion I used writeUTF, see https://issues.apache.org/jira/browse/HUDI-1443 for some context.